src/Sulu/Bundle/AdminBundle/Resources/config/services.xml

```diff
@@ -92,9 +92,17 @@
             <argument type="service" id="sulu_admin.form_metadata.form_metadata_mapper"/>
         </service>
 
+        <service
+            id="sulu_admin.property_metadata_mapper_registry"
+            class="Sulu\Bundle\AdminBundle\Metadata\SchemaMetadata\PropertyMetadataMapperRegistry">
+            <argument type="tagged_locator" tag="sulu_admin.property_metadata_mapper" index-by="type"/>
```

Comment:
not sure if there are any advantage/disadvantages, but we used `type="tagged"` with a `default-index-method` when implementing the list filter feature:
https://github.com/sulu/sulu/pull/5035/files#diff-634082d921f72e34116b5f1b8e0bc715867d2b93182f5b6ab90445caf74f02bdR33-R43

it would be nice to keep these things consistent for new features

Reply:
+1 for using the `default-index-method`, but that might be a bit confusing, since probably most `PropertyMetadataMapper`s are implemented in their corresponding content types.

About the `type="tagged"`, I would prefer keeping the `tagged_locator` for performance reasons, because the `ServiceLocator` is a lazy service container.

WDYT?

Reply:
Does it really have performance advantages? I think Symfony only creates that service when it is used (isn't that the advantage of the container being split since Symfony 3.4(?))? It might be a bit faster if there are loads of `PropertyMetadataMapper` registered, but I could imagine that it is slower if most of the `PropertyMetadataMapper`s are used (which I guess will be the case, especially in the beginning), because the laziness comes with some overhead.

That being said, I don't think that consistency is super important here, and if it brings a performance advantage I would keep it this way.

Regarding the `default-index-method`: That approach really gets a bit problematic, if we add more interfaces to the `ContentType`, at some point there might be naming collisions. But if we say we want to split the `ContentType` classes into multiple classes in the future, it would be fine for me to do it this way at the moment.

Reply:
Ok, probably you're right, that using a `ServiceLocator` won't have performance advantages with so few services, but it's also easier to work with in the `PropertyMetadataMapperRegistry`, because the `ServiceLocator` already comes with exactly those methods the `PropertyMetadataMapperRegistry` also has ...

Because the `ContentType` will probably be removed in `3.0`, we can assume there will be multiple classes per field type in future, so I guess naming collisions shouldn't be a problem

Reply:
@nnatter would you be okay with that?

Reply:
yeah sure - as far as i understand `type="tagged"` and `type="tagged_locator"` does not make any difference for the developer anyway 🙂

Reply:
exactly, it just simplifies the `PropertyMetadataMapperRegistry`